### PR TITLE
add failover policy to service resolver and proxy default

### PR DESCRIPTION
**Pull request by malizz**
add failover policy to service resolver and proxy default
ticket: https://hashicorp.atlassian.net/browse/NET-2851

Checklist:
- [x] Tests added
- [ ] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

File: charts/consul/templates/crd-proxydefaults.yaml [hunk 1]
```diff
@@ -143,6 +143,14 @@ spec:
                       type: object
                     type: array
                 type: object
+              failoverPolicy: 
+                description: FailoverPolicy specifies the exact mechanism used for failover.
+                properties:
+                  mode:
+                    description: Mode specifies the type of failover that will be performed. 
+                      Valid values are "default", "" (equivalent to "default") and "order-by-locality".
```
Comment (erichaberkorn) [inline charts/consul/templates/crd-proxydefaults.yaml:151]: I'm currently changing the name of the "default" case to "sequential" in this PR https://github.com/hashicorp/consul-enterprise/pull/4734/files#diff-dadef19b7fe31571431416cb4db25634e4f83cad2b9547e543fa0277e8b4f221R1437

**Review by erichaberkorn — COMMENTED** [review]
@wilkermichael will have more context on this because he is more familiar with the consul-k8s codebase.

File: charts/consul/templates/crd-proxydefaults.yaml (hunk 1 above)
Comment (erichaberkorn) [reply charts/consul/templates/crd-proxydefaults.yaml:151]: I just remembered that we are adding the regions field in that PR too

Comment (malizz) [reply charts/consul/templates/crd-proxydefaults.yaml]: Thanks, updated the value, and will add the extra field when api package is updated.

File: control-plane/go.sum
Comment (wilkermichael) [inline control-plane/go.sum:1]: I had to run `go mod tidy` after pulling to get things working, so just try running that locally and committing the changes

**Review by wilkermichael — COMMENTED** [review]
Great work Mali! Sorry I took so long to get to this

Some minor comments, also can you let me know if you tested things locally and update the PR under the `how I tested section` thanks!

File: charts/consul/templates/crd-proxydefaults.yaml (hunk 1 above)
Comment (malizz) [reply charts/consul/templates/crd-proxydefaults.yaml]: No worries. Thanks!

File: control-plane/connect-inject/controllers/endpoints/endpoints_controller_ent_test.go [hunk 2]
```diff
@@ -2107,7 +2111,8 @@ func TestReconcileDeleteGatewayWithNamespaces(t *testing.T) {
 
 				if tt.enableACLs {
 					_, _, err = consulClient.ACL().TokenRead(token.AccessorID, nil)
-					require.EqualError(t, err, "Unexpected response code: 403 (ACL not found)")
+					require.EqualError(t, err,
```
Comment (wilkermichael) [inline control-plane/connect-inject/controllers/endpoints/endpoints_controller_ent_test.go:2114]: I have a suggestion to potentially make these tests less brittle for the future. 

Can we switch all of the ACL error string matches to just check for `contains` like
`require.Contains(t, err, "ACL not found")` or something like that

**Review by wilkermichael — APPROVED** [review]
Great work Mali, thanks for fixing up all the tests and getting this running with 1.16-dev!

File: control-plane/connect-inject/controllers/endpoints/endpoints_controller_ent_test.go (hunk 2 above)
Comment (malizz) [reply control-plane/connect-inject/controllers/endpoints/endpoints_controller_ent_test.go:2114]: Will do, the reason I didn't change them all to this assertion was to stay precise (if it actually matters to be able to distinguish between the two cases). Thanks!

**Pull request merged**